rp2040-hal/src/gpio/reg.rs

```diff
@@ -255,6 +255,101 @@ pub(super) unsafe trait RegisterInterface {
         }
     }
 
+    /// Clear interrupt.
+    #[inline]
+    fn clear_interrupt(&self, interrupt: Interrupt) {
+        let num = self.id().num as usize;
+        unsafe {
+            let io = &(*pac::IO_BANK0::ptr());
+            let reg = (&io.intr0).as_ptr().add(num / 8);
+            let bit_in_reg = num % 8 * 4 + interrupt as usize;
+            *reg |= 1 << bit_in_reg;
+        }
+    }
+
+    /// Interrupt status.
+    #[inline]
+    fn interrupt_status(&self, interrupt: Interrupt) -> bool {
+        let num = self.id().num as usize;
+        unsafe {
+            let cpuid = *(pac::SIO::ptr() as *const u32);
+            let io = &(*pac::IO_BANK0::ptr());
+            let reg = (&io.proc0_ints0)
+                .as_ptr()
+                .add(num / 8 + cpuid as usize * 12);
```

Comment:
These pointers are likely to be unsound (due to pointing outside of the original object).
I'm not going to block merging based on this, but I'll raise an issue so we can fix up the PAC to make these safe